### PR TITLE
Fix image download headers regresison and fix png image issue

### PR DESCRIPTION
**Pull request by wanliAlex**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
bug fix

* **What is the current behavior?** (You can also link to an open issue here)
This is a regression as image_download_headers. Private images can not be downloaded even if users passes image_download_headers. Thi is a regression.

We also find an issue that when providing weighted queries, Marqo can not accept a queries with more than 2 modalities. An 500 error is raised:
```python
res = mq.index("test").search({"test": 1, "https://marqo-k400-video-test-dataset.s3.amazonaws.com/videos/-LenAZs2OFM_000002_000012.mp4": 1, "https://raw.githubusercontent.com/marqo-ai/marqo/mainline/examples/ImageSearchGuide/data/image1.jpg": 1})


Languagebind models can't take `.png` images as input.
```

* **What is the new behavior (if this is a feature change)?**
- We introduce a new parameter, `mediaDownloadHears` and will deprecate `imageDownloadHeaders` in the future.
- We fix the regression. Users can add documents with private images by providing media_download_headers.
- Fix the bug that users can not provide more than 2 modalities in the weighted query.
- Fix the bug that languagebind models can't take `.png` images as input. It can not index `.png` models correctly.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)

* **Related Python client changes** (link commit/PR here)

* **Related documentation changes** (link commit/PR here)

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/api/models/add_docs_objects.py [hunk 1]
```diff
@@ -38,3 +39,22 @@ def validate_thread_counts(cls, values):
...
+        If imageDownloadHeaders is set, set mediaDownloadHeaders to it and use mediaDownloadHeaders in the
+        rest of the code.
+
+        imageDownloadHeaders is deprecated and will be removed in the future.
+        """
+        image_download_headers = values.get('imageDownloadHeaders')
+        media_download_headers = values.get('mediaDownloadHeaders')
+        if image_download_headers and media_download_headers:
+            raise ValueError("Cannot set both imageDownloadHeaders and mediaDownloadHeaders. "
+                             "The imageDownloadHeaders is deprecated and will be removed in the future. "
```
Comment (farshidz) [inline src/marqo/api/models/add_docs_objects.py]: "'imageDownloadHeaders' is deprecated and will be removed in the future" (no The, add single quotes). 

Or "The 'imageDownloadHeaders' parameter is deprecated and will be removed in the future."

I prefer the first one.

File: src/marqo/api/models/embed_request.py [hunk 2]
```diff
@@ -6,18 +6,26 @@
...
-    image_download_headers: Optional[Dict] = None
+    image_download_headers: Optional[Dict] = Field(default=None, alias="imageDownloadHeaders")
+    mediaDownloadHeaders: Optional[Dict] = None
     modelAuth: Optional[ModelAuth] = None
-    content_type: Optional[EmbedContentType] = EmbedContentType.Query
+    content_type: Optional[EmbedContentType] = Field(default=EmbedContentType.Query, alias="contentType")
+
+    @root_validator(pre=True)
+    def _test(cls, values):
+        print(values)
```
Comment (farshidz) [inline src/marqo/api/models/embed_request.py]: ?

File: src/marqo/api/models/embed_request.py [hunk 3]
```diff
@@ -47,4 +55,23 @@ def validate_content(cls, value):
...
+
+        If imageDownloadHeaders is set, set mediaDownloadHeaders to it and use mediaDownloadHeaders in the
+        rest of the code.
+
+        imageDownloadHeaders is deprecated and will be removed in the future.
+        """
+        image_download_headers = values.get('imageDownloadHeaders')
+        media_download_headers = values.get('mediaDownloadHeaders')
+        if image_download_headers and media_download_headers:
+            raise ValueError("Cannot set both imageDownloadHeaders and mediaDownloadHeaders. "
```
Comment (farshidz) [inline src/marqo/api/models/embed_request.py:67]: See my other comment re grammar and single quote wrapping

File: src/marqo/s2_inference/clip_utils.py [hunk 4]
```diff
@@ -485,7 +487,7 @@ def encode(self, inputs: Union[str, ImageType, List[Union[str, ImageType]]],
 
         if is_image:
             logger.debug('image')
-            image_download_headers = kwargs.get("image_download_headers", dict())
+            image_download_headers = kwargs.get("media_download_headers", dict())
```
Comment (farshidz) [inline src/marqo/s2_inference/clip_utils.py]: There is a reference to `image_download_headers` above. Are we being consistent with which one we expect here? Is snake case the expected input?

File: src/marqo/s2_inference/multimodal_model_load.py [hunk 5]
```diff
@@ -109,29 +110,37 @@ def preprocessor(self, modality):
...
+def fetch_content_sample(url, media_download_headers: Optional[dict] = None, sample_size=10240):  # 10 KB
+    # It's ok to pass None to requests.get() for headers and it won't change the default headers
+    """Fetch a sample of the content from the URL.
+
+    Raises:
+        HTTPError: If the response status code is not 200
+    """
+    response = requests.get(url, stream=True, headers=media_download_headers)
+    if response.status_code != 200:
+        response.raise_for_status()
```
Comment (farshidz) [inline src/marqo/s2_inference/multimodal_model_load.py]: The 200 if should be already covered by `response.raise_for_status` so it redundant I think

File: src/marqo/tensor_search/models/search.py [hunk 6]
```diff
@@ -75,4 +77,29 @@ def __init__(self, **data):
...
\ No newline at end of file
+        return v
+
+
+class QueryContent(BaseModel):
+    content: str
+    modality: Modality
+
+
+class QueryContentCollector(BaseModel):
```
Comment (farshidz) [inline src/marqo/tensor_search/models/search.py:88]: Why don't we call this just Query?

File: src/marqo/tensor_search/tensor_search.py [hunk 7]
```diff
@@ -2019,19 +2025,20 @@ def add_prefix_to_queries(queries: List[BulkSearchQueryEntity]) -> List[BulkSear
         if q.q is None:
             prefixed_q = q.q
         elif isinstance(q.q, str):
-            if _is_image(q.q):
-                prefixed_q = q.q
-            else:
+            modality = infer_modality(q.q, q.mediaDownloadHeaders)
```
Comment (farshidz) [inline src/marqo/tensor_search/tensor_search.py:2028]: Is this the same logic? We don't want to change image modality inference logic

File: src/marqo/tensor_search/tensor_search.py [hunk 8]
```diff
@@ -2589,7 +2596,7 @@ def vectorise_multimodal_combination_field_structured(
                     content=prefixed_text_content,
                     device=device,
                     normalize_embeddings=normalize_embeddings,
-                    infer=False,
+                    infer=True,
```
Comment (farshidz) [inline src/marqo/tensor_search/tensor_search.py]: What does this change do?

File: tests/tensor_search/integ_tests/test_add_documents_combined.py [hunk 9]
```diff
@@ -1092,4 +1093,62 @@ def test_textIndexEmbeddingsUnnormalized(self):
...
+                    )
+                )
+                self.assertTrue(res.errors)
+                items = res.items
+                self.assertEqual(2, len(items))
+                for item in items:
+                    self.assertEqual(400, item.status)
+                    self.assertIn("403", item.message)
+
+    def test_add_private_images_success(self):
```
Comment (farshidz) [inline tests/tensor_search/integ_tests/test_add_documents_combined.py:1130]: Are we not testing private video and audio? Also tests more multiple modalities in search which this PR also addresses. I think we need to api test these as well

File: src/marqo/api/models/add_docs_objects.py (hunk 1 above)
Comment (wanliAlex) [reply src/marqo/api/models/add_docs_objects.py]: done

File: src/marqo/api/models/embed_request.py (hunk 2 above)
Comment (wanliAlex) [reply src/marqo/api/models/embed_request.py]: removed

File: src/marqo/api/models/embed_request.py (hunk 3 above)
Comment (wanliAlex) [reply src/marqo/api/models/embed_request.py:67]: done

File: src/marqo/s2_inference/multimodal_model_load.py (hunk 5 above)
Comment (wanliAlex) [reply src/marqo/s2_inference/multimodal_model_load.py]: done

File: src/marqo/tensor_search/models/search.py (hunk 6 above)
Comment (wanliAlex) [reply src/marqo/tensor_search/models/search.py:88]: I feel Query is a more generic word for this

File: src/marqo/tensor_search/tensor_search.py (hunk 7 above)
Comment (wanliAlex) [reply src/marqo/tensor_search/tensor_search.py:2028]: Case 1: q is a string, then it will be categorized as text in both cases
Case 2: q is a valid URL, it will be categorized as an image in both cases
Case 3: q is an invalid URL, it will be categorized as an image before and raise an error during downloading. Now, we raise the error when inferring the modality.

Therefore I think the behaviour is the same.

File: src/marqo/tensor_search/tensor_search.py (hunk 8 above)
Comment (wanliAlex) [reply src/marqo/tensor_search/tensor_search.py]: Reverted

File: tests/tensor_search/integ_tests/test_add_documents_combined.py (hunk 9 above)
Comment (wanliAlex) [reply tests/tensor_search/integ_tests/test_add_documents_combined.py:1130]: Yes I am adding more tests.

File: src/marqo/s2_inference/clip_utils.py (hunk 4 above)
Comment (wanliAlex) [reply src/marqo/s2_inference/clip_utils.py]: A mixture of usage is confusing. I was using minimum efforts to get this PR merged. 
Now I have changed all the image_download_headers to media_download_headers.

File: tests/marqo_test.py [hunk 10]
```diff
@@ -36,6 +36,21 @@ class TestImageUrls(str, Enum):
...
+class TestAudioUrls(str, Enum):
+    __test__ = False
+    AUDIO1 = "https://marqo-ecs-50-audio-test-dataset.s3.us-east-1.amazonaws.com/audios/1-100032-A-0.wav"
+    AUDIO2 = "https://marqo-ecs-50-audio-test-dataset.s3.us-east-1.amazonaws.com/audios/1-115545-C-48.wav"
+    AUDIO3 = "https://marqo-ecs-50-audio-test-dataset.s3.us-east-1.amazonaws.com/audios/1-119125-A-45.wav"
+
+
+class TestVideoUrls(str, Enum):
+    __test__ = False
+    VIDEO1 = "https://marqo-k400-video-test-dataset.s3.us-east-1.amazonaws.com/videos/--_S9IDQPLg_000135_000145.mp4"
```
Comment (farshidz) [inline tests/marqo_test.py:48]: Are these datasets we're hosting publicly?

File: tests/tensor_search/integ_tests/test_search_combined.py [hunk 11]
```diff
@@ -1011,4 +1010,150 @@ def test_search_over_private_images_with_media_download_headers(self):
                         text=query,
                         search_method=SearchMethod.TENSOR,
                         media_download_headers={"marqo_media_header": "media_header_test_key"}
+                    )
+
+    def test_lexical_search_DoesNotErrorWithEscapedQuotes(self):
```
Comment (farshidz) [inline tests/tensor_search/integ_tests/test_search_combined.py]: This doesn't seem to belong in this PR?

Comment (wanliAlex) [reply tests/tensor_search/integ_tests/test_search_combined.py]: They are not added by me

File: tests/tensor_search/integ_tests/test_search_combined.py [hunk 12]
```diff
@@ -1012,3 +1057,103 @@ def test_lexical_search_DoesNotErrorWithEscapedQuotes(self):
                         )
                         self.assertEqual(len(expected_ids), len(res['hits']))
                         self.assertEqual(set(expected_ids), {hit['_id'] for hit in res['hits']})
+
+
+@pytest.mark.largemodel
```
Comment (wanliAlex) [inline tests/tensor_search/integ_tests/test_search_combined.py]: @farshidz New tests here

File: tests/marqo_test.py (hunk 10 above)
Comment (wanliAlex) [reply tests/marqo_test.py:48]: Yes, on s3

**Review by farshidz — APPROVED** [review]
